### PR TITLE
Config Reader

### PR DESCRIPTION
**Pull request by ripecosta**
## What does this change?
Adds a Config Reader and Failure class to the repo.

### CODE REVIEW

File: src/main/scala/payment_failure_comms/models/Failure.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package payment_failure_comms.models
+
+sealed trait Failure {
+  def kind: String
+  def details: String
+}
+
+case class ConfigFailure(details: String) extends Failure {
+  val kind: String = "Config"
+}
```
Comment (johnduffell) [inline src/main/scala/payment_failure_comms/models/Failure.scala:10]: if there's only one for now, consider just making a standalone case class for now and add the trait later if/when it's needed

File: src/main/scala/payment_failure_comms/models/Config.scala [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package payment_failure_comms.models
+
+case class Config(braze: BrazeConfig)
```
Comment (johnduffell) [inline src/main/scala/payment_failure_comms/models/Config.scala:3]: is the outer wrapper needed (at this stage)? or just to encourge people to structure it sensibly when it is expanded?

**Review by johnduffell — APPROVED** [review]
looks good, it's hard to tell without the context whether "yagni *" should apply, but I added some comments anyway

* https://martinfowler.com/bliki/Yagni.html

File: src/main/scala/payment_failure_comms/models/Failure.scala (hunk 1 above)
Comment (ripecosta) [reply src/main/scala/payment_failure_comms/models/Failure.scala:10]: Other Failure case classes are added in the following PRs (a couple of those are already up). I knew it was going to need the trait so I added it in from the start.

**Pull request merged**